### PR TITLE
G-Mode Gauntlet

### PR DESCRIPTION
**Pull request by kjbranch**

### CODE REVIEW

File: region/crateria/west.json [hunk 1]
```diff
@@ -1389,6 +1389,38 @@
...
+                  "requires": [
+                    {"or": [
+                      "h_canDestroyBombWalls",
+                      {"canComeInCharged": {
+                        "fromNode": 1,
+                        "framesRemaining": 180,
+                        "shinesparkFrames": 0
+                      }},
+                      {"and": [
+                        {"comeInWithGMode": {
```
Comment (osse101) [inline region/crateria/west.json:1408]: Seems extremely niche
I don't know that there is a reason you'd want to use GMode to get GMode on this door.

File: region/crateria/west.json [hunk 2]
```diff
@@ -1389,6 +1389,38 @@
...
+                          "artificialMorph": true
+                        }},
+                        {"or": [
+                          "Bombs",
+                          {"ammo": { "type": "PowerBomb", "count": 1 }}
+                        ]}
+                      ]}
+                    ]}
+                  ],
+                  "devNote": "FIXME: It is not currently possible to represent coming in from the opposite door blue or in an artificial morph."
```
Comment (osse101) [inline region/crateria/west.json:1420]: Artifical morph can't overload PLMs to cross the room. Bombing blocks would make them solid.

File: region/crateria/west.json [hunk 3]
```diff
@@ -1468,6 +1500,38 @@
...
+                  "requires": [
+                    {"or": [
+                      "h_canDestroyBombWalls",
+                      {"canComeInCharged": {
+                        "fromNode": 2,
+                        "framesRemaining": 180,
+                        "shinesparkFrames": 0
+                      }},
+                      {"and": [
+                        {"comeInWithGMode": {
```
Comment (osse101) [inline region/crateria/west.json:1519]: Same comment.  You could just walk out the door in GMode

File: region/crateria/west.json (hunk 1 above)
Comment (kjbranch) [reply region/crateria/west.json:1408]: To get into the adjacent room in direct gmode

Comment (osse101) [reply region/crateria/west.json:1408]: Should probably add a crystal flash to the requirements. As a new strat, really.
Artifical morph->CF->end GMode so the PB breaks the blocks->damage down on waver to appropriate health->leaveWithGModeSetup

File: region/crateria/west.json [hunk 4]
```diff
@@ -1403,21 +1403,24 @@
...
+                  "notable": false,
+                  "requires": [
+                    {"comeInWithGMode": {
+                      "fromNodes": [1],
+                      "mode": "any",
+                      "artificialMorph": true
+                    }},
+                    "h_canArtificialMorphCrystalFlash"
+                  ],
+                  "note": "Place to PB to free but not kill the Waver. After using CF, lower back down using the acid or the Waver.",
```
Comment (osse101) [inline region/crateria/west.json:1422]: ~~to~~ the

File: region/crateria/west.json [hunk 5]
```diff
@@ -1514,21 +1517,24 @@
...
+                  "notable": false,
+                  "requires": [
+                    {"comeInWithGMode": {
+                      "fromNodes": [2],
+                      "mode": "any",
+                      "artificialMorph": true
+                    }},
+                    "h_canArtificialMorphCrystalFlash"
+                  ],
+                  "note": "Place to PB to free but not kill the Waver. After using CF, lower back down using the acid or the Waver.",
```
Comment (osse101) [inline region/crateria/west.json:1536]: ~~to~~ the

**Review by osse101 — DISMISSED** [review]

**Pull request merged**